lib/js/components/Tile/Tile.vue

```diff
@@ -370,68 +406,58 @@ $tile-colors: (
 		}
 	}
 
-	&__iconLeft {
-		margin-right: $space-xs;
-	}
-
-	&__iconRight {
-		margin-left: $space-xs;
-
-		&.-ds-hiddenOnMobile {
-			display: none;
-
-			@media #{breakpoint-s()} {
-				display: block;
-			}
-		}
-	}
-
 	&.-ds-interactive:not(.-ds-disabled):not(.-ds-loading) {
 		cursor: pointer;
 	}
 }
 </style>
 
-<script lang="ts">
+<script setup lang="ts">
 import DsIcon, { ICON_SIZES, ICONS } from '../Icons/Icon';
-import { TILE_ADDITIONAL_TEXT_MAX_WIDTHS, TILE_COLORS, TILE_STATES } from './Tile.consts';
-import { props } from './Tile.shared';
-import { defineComponent } from 'vue';
-
-export default defineComponent({
-	name: 'Tile',
-	components: {
-		DsIcon,
-	},
-	props,
-	data() {
-		return {
-			ICONS: Object.freeze(ICONS),
-			ICON_SIZES: Object.freeze(ICON_SIZES),
-			TILE_STATES: Object.freeze(TILE_STATES),
-			TILE_ADDITIONAL_TEXT_MAX_WIDTHS: Object.freeze(TILE_ADDITIONAL_TEXT_MAX_WIDTHS),
-		};
-	},
-	computed: {
-		tileColor() {
-			return {
-				[TILE_COLORS.NEUTRAL]: '-ds-neutral',
-				[TILE_COLORS.NEUTRAL_WEAK]: '-ds-neutralWeak',
-				[TILE_COLORS.PRIMARY]: '-ds-primary',
-				[TILE_COLORS.SUCCESS]: '-ds-success',
-				[TILE_COLORS.FAIL]: '-ds-fail',
-				[TILE_COLORS.DANGER]: '-ds-danger',
-				[TILE_COLORS.WARNING]: '-ds-warning',
-				[TILE_COLORS.INFO]: '-ds-info',
-			}[this.color];
-		},
-		tileState() {
-			return {
-				[TILE_STATES.DEFAULT]: null,
-				[TILE_STATES.DISABLED]: '-ds-disabled',
-				[TILE_STATES.LOADING]: '-ds-loading',
-			}[this.state];
-		},
-	},
+import {
+	TILE_ADDITIONAL_TEXT_MAX_WIDTHS,
+	TILE_COLORS,
+	TILE_COMPACT_LAYOUTS,
+	TILE_STATES,
+} from './Tile.consts';
+import { props as TileProps } from './Tile.shared';
+import { computed } from 'vue';
+
+const props = defineProps(TileProps);
+
+const tileCompactLayoutClass = computed(() => {
+	switch (props.compactLayout) {
+		case TILE_COMPACT_LAYOUTS.ICON_LEFT_ONLY:
+			return '-ds-iconLeftOnly';
+		case TILE_COMPACT_LAYOUTS.ICON_RIGHT_ONLY:
+			return '-ds-iconRightOnly';
+		case TILE_COMPACT_LAYOUTS.ICON_RIGHT_HIDDEN:
+			return '-ds-iconRightHidden';
+		default:
+			return false;
```

Comment:
I wonder if that would work, but perhaps if you return `''` instead of `false` you could remove the `!!` casting in template and use `tileCompactLayoutClass` directly in `:class=[...]` binding.